ehitchhiking-ui/src/app/shared/components/rate-passengers-modal/rate-passengers-modal.component.ts

```diff
@@ -2,54 +2,116 @@ import {Component, Inject, OnInit} from '@angular/core';
 import {MAT_DIALOG_DATA} from '@angular/material';
-import {User} from '@shared/components/rate-passengers-modal/user';
 import {UserState} from '@shared/enums/UserState';
+import {RatePassengersApiService} from '@shared/services/api.services/rate-passengers-api.service';
+import {LoaderSize} from '@shared/enums/pre-loader-sizes';
+import {BlacklistedUser} from '@shared/components/rate-passengers-modal/user';
 
 @Component({
 	selector: 'rate-passengers-modal',
 	templateUrl: './rate-passengers-modal.component.html',
 	styleUrls: ['./rate-passengers-modal.component.sass'],
 })
 export class RatePassengersModalComponent implements OnInit {
-	users: User[] = [];
+	users: RatedUser[] = [];
 	UserState = UserState;
-
+	idTripDriver = 12;
+	idTripPassenger = 4;
+	loading = true;
+	loaderSize: LoaderSize = LoaderSize.Large;
 	constructor(
 		public dialogRef: MatDialogRef<RatePassengersModalComponent>,
-		@Inject(MAT_DIALOG_DATA) public data: any
+		@Inject(MAT_DIALOG_DATA) public data: any,
+		private apiRatePassengersService: RatePassengersApiService
 	) {}
 
 	ngOnInit(): void {
-		// here should be just GET 'info' about trip
-		if (this.data.dataKey === UserState.Driver) {
-			this.users = [
-				{id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false},
-				{id: 1, name: 'Egor', rating: 0, isAddedToBlackList: false},
-				{id: 2, name: 'Semen', rating: 0, isAddedToBlackList: false},
-				{id: 3, name: 'Fedor', rating: 0, isAddedToBlackList: false},
-			];
-		} else {
-			this.users = [{id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false}];
+		switch (this.data.dataKey) {
+			case UserState.Passenger:
+				this.loadDriversList();
+				break;
+			case UserState.Driver:
+				this.loadPassengersList();
+				break;
+			default:
+				break;
 		}
 	}
 
+	loadPassengersList(): void {
+		this.loading = true;
+		this.apiRatePassengersService
+			.getTripPassengers(this.idTripPassenger)
+			.then((data: RatedUser[]) => {
+				this.users = data;
+				console.log(data);
+			})
+			.finally(() => {
+				this.loading = false;
+			});
+	}
+
+	loadDriversList(): void {
+		this.loading = true;
+		this.apiRatePassengersService
+			.getTripDriver(this.idTripDriver)
+			.then((data: RatedUser[]) => {
+				this.users = data;
+				console.log(data);
+			})
+			.finally(() => {
+				this.loading = false;
+			});
+	}
+
 	rateUser(clickObj: StarClickMeta): void {
-		const user = this.users.find((i: User) => i.id === clickObj.itemId);
-		if (!!user) {
-			user.rating = clickObj.rating;
-		}
+		this.users.forEach((i: RatedUser) => {
+			i.id === clickObj.itemId ? (i.rate = clickObj.rating) : '';
+			return i;
+		});
 	}
 
 	exitTrip(): void {
-		this.dialogRef.close();
+		const users = [];
+		const blockedUsers = [];
+		this.users.forEach((user) => {
+			users.push({
+				id: user.id,
+				rate: user.rate,
+			});
+			blockedUsers.push({
+				id: user.id,
+				isBlocked: user.isBlocked,
+			});
+		});
+		let arr = [];
```

Comment:
const requests = submitChanges();

submitChanges(): Promise<any>[] {
return this.data.dataKey === UserState.Passenger
				? [
						this.apiRatePassengersService.addRateDriver(users),
						this.apiRatePassengersService.addBlacklistDriver(this.idTripDriver, blockedUsers),
				  ]
				: [
						this.apiRatePassengersService.addRatePassenger(users),
						this.apiRatePassengersService.addBlacklistPass(this.idTripPassenger, blockedUsers),
				  ];
}